Applications/Spire/Source/SpireTester/LocalOrderImbalanceIndicatorModelTester.cpp

```diff
@@ -0,0 +1,134 @@
+#include <doctest/doctest.h>
+#include <boost/date_time/posix_time/ptime.hpp>
+#include "Nexus/Definitions/Money.hpp"
+#include "Nexus/Definitions/OrderImbalance.hpp"
+#include "Nexus/Definitions/Quantity.hpp"
+#include "Nexus/Definitions/Security.hpp"
+#include "Nexus/Definitions/Side.hpp"
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/SpireTester/SpireTester.hpp"
+
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto make_imbalance(const auto& symbol, auto timestamp) {
+    return OrderImbalance(Security(symbol, DefaultCountries::US()), Side::ASK,
+      Quantity(100), Money(12.34), from_time_t(timestamp));
+  }
+
+  auto closed(auto lower, auto upper) {
+    return TimeInterval::closed(from_time_t(lower), from_time_t(upper));
+  }
+
+  auto open(auto lower, auto upper) {
+    return TimeInterval::open(from_time_t(lower), from_time_t(upper));
+  }
+
+  auto load_closed(const auto& model, auto lower, auto upper) {
+    auto result = model->subscribe(closed(lower, upper), [] (const auto&) {});
+    return wait(std::move(result.m_snapshot));
+  }
+
+  auto load_open(const auto& model, auto lower, auto upper) {
+    auto result = model->subscribe(open(lower, upper), [] (const auto&) {});
+    return wait(std::move(result.m_snapshot));
+  }
+
+  bool contains(const auto& container, const auto& value) {
+    return
+      std::find(container.begin(), container.end(), value) != container.end();
+  }
+}
+
+TEST_SUITE("LocalOrderImbalanceIndicatorModel") {
```

Comment:
Add a unit test to load historical imbalances, along the lines of:

auto i1 = make_imbalance("A", 100);
auto i2 = make_imbalance("A", 200);
auto load1 = load_closed(model, 50, 150);
REQUIRE(load1.size() == 1 && load1.front() == i1);
auto load2 = load_closed(model, 150, 250);
REQUIRE(load2.size() == 1 && load2.front() == i2);